logger/index.ts

```diff
@@ -1,45 +1,97 @@
 import format from "@arcjet/sprintf";
 
-export type LogLevel = "DEBUG" | "LOG" | "WARN" | "ERROR";
-
-function getEnvLogLevel(): LogLevel | undefined {
-  const level = process.env["ARCJET_LOG_LEVEL"];
-  switch (level) {
-    case "DEBUG":
-    case "LOG":
-    case "WARN":
-    case "ERROR":
-      return level;
-    default:
-      return undefined;
+function bigintReplacer(key: string, value: unknown) {
+  if (typeof value === "bigint") {
+    return "[BigInt]";
+  }
+
+  return value;
+}
+
+// TODO: Deduplicate this and sprintf implementation
+function tryStringify(o: unknown) {
+  try {
+    return JSON.stringify(o, bigintReplacer);
+  } catch (e) {
+    return "[Circular]";
   }
 }
 
+export type LogLevel = "debug" | "info" | "warn" | "error";
+
+export interface LoggerOptions {
+  level: LogLevel;
+}
+
+const PREFIX = "✦Aj";
```

Comment:
The logs would look nicer if this were `✦AJ` because we know print `✦Aj INFO my message` to align closer to pino-pretty. But I know the brand guidelines say we are supposed to use the lowercase `j`.

cc @davidmytton @morekid

Reply:
`✦Aj INFO` vs `✦AJ INFO `

I see your point alignment-wise, but the current version is more readable in general (full caps is usually less so, and "AJ" in particular looks a bit jarring), also easier to understand as "Arcjet" abbreviation, and preserves its recognizability when encountered elsewhere.

Reply:
we could consider just having the unicode `✦` eventually for all logs if that gets in the way? not sure it'll be enough to differentiate and make them look like Arcjet logs specifically.

Reply:
> not sure it'll be enough to differentiate and make them look like Arcjet logs specifically.

I think we need some sort of differentiator, like `Aj`. I just think `Aj INFO` looks weird because `INFO`/`WARN`/`ERROR` are fully capitalized.

Reply:
I wonder if this is a scenario where an exception to the case rule makes sense. We're Arcjet, not ArcJet, so normally I'd say `Aj` is the correct casing. But in this context where the whole prefix to the log is upper case, `Aj` looks a bit weird. Does the context overrule being consistent?

Reply:
I think it's decided that we're just going to keep the lowercase j so the logs look like:

Notice that I did add 2 extra spaces so the object properties wouldn't be exactly aligned with `DEBUG`
```
✦Aj DEBUG Decide request to https://decide.arcjet.orb.local:4082
✦Aj DEBUG Decide response
      id: "req_01hzpy6jz7ftn8ajsn2kdzkdbe"
      fingerprint: "fp_1_3e48ef9d22e096da6838540fb846999890462c8a32730a4f7a5eaee6945315f7"
      path: "/api/arcjet"
      runtime: "node"
      ttl: 0
      conclusion: "ALLOW"
      reason: {"type":"SHIELD","shieldTriggered":false}
      ruleResults: [{"ruleId":"","ttl":0,"state":"RUN","conclusion":"ALLOW","reason":{"type":"SHIELD","shieldTriggered":false}}]
✦Aj LATENCY decideApi: 121.4ms
✦Aj LATENCY remote: 121.436ms
```

Reply:
@blaine-arcjet looks good to me, indentation as well, ✦ftw!